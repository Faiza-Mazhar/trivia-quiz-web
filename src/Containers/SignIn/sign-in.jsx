import { useState } from "react";
import CustomButton from "../../Components/CustomButton/custom-button";
import FormInput from "../../Components/FormInput/form-input";
import GoogleLogo from "../../assets/googleLogo.png";
import { auth, signInWithGoogle } from "../../firebase/firebase.utils";

import "./sign-in.style.scss";
const SignIn = ({ setIsSignIn }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = async (event) => {
    event.preventDefault();

    try {
      const { user } = await auth.signInWithEmailAndPassword(email, password);
      console.log({ user });
      setEmail("");
      setPassword("");
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <div className="sign-in">
      <div className="title">
        <span> Sign in with your email and password.</span>
      </div>

      <form onSubmit={handleSubmit}>
        <FormInput
          name="email"
          type="email"
          handleChange={(event) => setEmail(event.target.value)}
          value={email}
          label="Email"
          required
        />

        <FormInput
          name="password"
          type="password"
          handleChange={(event) => setPassword(event.target.value)}
          value={password}
          label="Password"
          required
          autoComplete="on"
        />
        <div className="buttons-container">
          <CustomButton type="submit">SIGN IN</CustomButton>
          <CustomButton onClick={signInWithGoogle} isGoogleSignIn type="button">
            <img src={GoogleLogo} alt="google-logo" width="30" height="30" />
            SIGN IN WITH GOOGLE
          </CustomButton>

          <label className="new-user-label" onClick={() => setIsSignIn(false)}>
            New User?
          </label>
        </div>
      </form>
    </div>
  );
};

export default SignIn;
