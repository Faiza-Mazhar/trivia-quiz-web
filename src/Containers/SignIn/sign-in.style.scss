.sign-in-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #abb899;
}

.title {
  margin: 10px 0px;
}

.buttons-container {
  display: flex;
  flex-direction: column;
  gap: 10px;
  justify-content: space-between;

  .new-user-label {
    padding: 10px;
    width: 100%;
    text-align: right;
    cursor: pointer;
    font-weight: bold;
  }
}
