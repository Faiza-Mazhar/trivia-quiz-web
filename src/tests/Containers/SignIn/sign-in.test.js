import renderer from "react-test-renderer";

import SignIn from "../../../Containers/SignIn/sign-in";

test("Sign form renders with correct props", () => {
  const component = renderer.create(<SignIn />).toJSON();
  expect(component).toMatchSnapshot();
});
