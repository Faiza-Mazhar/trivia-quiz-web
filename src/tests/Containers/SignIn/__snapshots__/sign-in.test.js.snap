// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign form renders with correct props 1`] = `
<div
  className="sign-in"
>
  <div
    className="title"
  >
    <span>
       Sign in with your email and password.
    </span>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <label
        className=" form-input-label"
      >
        Email
      </label>
      <input
        className="form-input"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="group"
    >
      <label
        className=" form-input-label"
      >
        Password
      </label>
      <input
        autoComplete="on"
        className="form-input"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="buttons-container"
    >
      <button
        className=" custom-button"
        type="submit"
      >
        SIGN IN
      </button>
      <button
        className="google-sign-in custom-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="google-logo"
          height="30"
          src="googleLogo.png"
          width="30"
        />
        SIGN IN WITH GOOGLE
      </button>
      <label
        className="new-user-label"
        onClick={[Function]}
      >
        New User?
      </label>
    </div>
  </form>
</div>
`;
